### PR TITLE
MDSmarines

### PR DESCRIPTION
**Pull request by Dreyal**
MDS changed for marines. 
Healspray and arcs are buffed by 15% too.

### CODE REVIEW

File: src/lua/CommunityBalanceMod/Combined/DamageTypes.lua [hunk 1]
```diff
@@ -60,48 +60,84 @@ function GetReceivesBiologicalDamage(entity)
 end
 
 local upgradedDamageScalars
-function NS2Gamerules_GetUpgradedDamageScalar( attacker, weaponTechId )
+local upgradedDamageScalarsStructure --MDS Marines
+function NS2Gamerules_GetUpgradedDamageScalar( attacker, weaponTechId, structureDamage ) --MDS added structureDamage
```
Comment (adsfgg) [inline src/lua/CommunityBalanceMod/Combined/DamageTypes.lua:64]: Do we need to update the other places where this function is called to include the new `structureDamage` argument? (FireMixin.lua and HitSounds.lua)

File: src/lua/CommunityBalanceMod/Combined/DamageTypes.lua [hunk 2]
```diff
@@ -60,48 +60,84 @@ function GetReceivesBiologicalDamage(entity)
...
+    end
+    
     return 1.0
 
 end
 
+-- MDS Marines
 -- Use this function to change damage according to current upgrades
-function NS2Gamerules_GetUpgradedDamage(attacker, doer, damage)
+function NS2Gamerules_GetUpgradedDamage(target, attacker, doer, damage)
```
Comment (adsfgg) [inline src/lua/CommunityBalanceMod/Combined/DamageTypes.lua:130]: Same again but for `target`

**Review by adsfgg — CHANGES_REQUESTED** [review]

File: src/lua/CommunityBalanceMod/Combined/DamageTypes.lua (hunk 2 above)
Comment (Dreyal) [reply src/lua/CommunityBalanceMod/Combined/DamageTypes.lua:130]: It only gets called in DamageType.lua
This should not cause any issues, but to be make sure its not getting called wrong by other mods we will append target to the last unused position of the function.

File: src/lua/CommunityBalanceMod/Combined/DamageTypes.lua (hunk 1 above)
Comment (Dreyal) [reply src/lua/CommunityBalanceMod/Combined/DamageTypes.lua:64]: I added the right call for FireMixin which actually increased flamethrowers burn damage like intended now. 
Hitsounds is irrelevant due to only player causing a hitsound

Comment (adsfgg) [reply src/lua/CommunityBalanceMod/Combined/DamageTypes.lua]: Perfect 👍

File: src/lua/CommunityBalanceMod/Combined/DamageTypes.lua (hunk 2 above)
Comment (adsfgg) [reply src/lua/CommunityBalanceMod/Combined/DamageTypes.lua]: Sounds good 👍

**Review by adsfgg — APPROVED** [review]

**Review by adsfgg — DISMISSED** [review]

**Comment by Dreyal** [comment]
Dev branch changed due to the previous merge, which requires us to review this merge again. Love it

**Review by adsfgg — APPROVED** [review]

**Pull request merged**